packages/daemon/src/host.js

```diff
@@ -430,51 +329,98 @@ export const makeHostMaker = ({
       return value;
     };
 
+    /**
+     * Attempts to introduce the given names to the specified party. The party in question
+     * must be incarnated before this function is called.
+     *
+     * @param {string} formulaIdentifier - The party's formula identifier.
+     * @param {Record<string,string>} introducedNames - The names to introduce.
+     * @returns {Promise<void>}
+     */
+    const introduceNamesToParty = async (
+      formulaIdentifier,
+      introducedNames,
+    ) => {
+      /** @type {import('./types.js').Controller<any, any>} */
+      const controller =
+        provideControllerForFormulaIdentifier(formulaIdentifier);
+      const { petStore: newPetStore } = await controller.internal;
+      await Promise.all(
+        Object.entries(introducedNames).map(async ([parentName, childName]) => {
+          const introducedFormulaIdentifier =
+            petStore.identifyLocal(parentName);
+          if (introducedFormulaIdentifier === undefined) {
+            return;
+          }
+          await newPetStore.write(childName, introducedFormulaIdentifier);
+        }),
+      );
+    };
+
+    /**
+     * @param {'guest' | 'host'} formulaType - The party's formula type.
+     * @param {string} [petName] - The party's potential pet name.
+     */
+    const getNamedParty = (formulaType, petName) => {
+      if (petName !== undefined) {
+        const formulaIdentifier = petStore.identifyLocal(petName);
+        if (formulaIdentifier !== undefined) {
+          if (parseId(formulaIdentifier).type !== formulaType) {
+            throw new Error(
+              `Existing pet name does not designate a ${formulaType} powers capability: ${q(
+                petName,
+              )}`,
+            );
+          }
+
+          return {
+            formulaIdentifier,
+            value: /** @type {Promise<any>} */ (
+              provideControllerForFormulaIdentifier(formulaIdentifier).external
+            ),
+          };
+        }
+      }
+      return undefined;
+    };
+
+    /**
+     * @param {string} [petName] - The pet name of the party.
+     */
+    const getDeferredTasksForParty = petName => {
+      /** @type {import('./types.js').DeferredTasks<import('./types.js').PartyDeferredTaskParams>} */
+      const tasks = makeDeferredTasks();
+      if (petName !== undefined) {
+        tasks.push(identifiers =>
+          petStore.write(petName, identifiers.partyFormulaIdentifier),
+        );
```

Comment:
It might be worth starting to think now about how this impacts writing the pet name at the end of a pet name path.